conn.go

```diff
@@ -495,6 +518,16 @@ func (conn *Conn) sendReply(dest string, serial uint32, values ...interface{}) {
 	conn.outLck.RUnlock()
 }
 
+func (conn *Conn) defaultSignalAction(fn func(h *defaultSignalHandler, ch chan<- *Signal), ch chan<- *Signal) {
+	conn.signalLck.RLock()
+	defer conn.signalLck.RUnlock()
+	if !isDefaultSignalHandler(conn.signalHandler) {
```

Comment:
Wouldn't this require that all signal handlers implement signal delivery as a channel send. In some cases this is not what is desired.